runtime/lua/vim/lsp/callbacks.lua

```diff
@@ -195,6 +195,25 @@ M['textDocument/peekDefinition'] = function(_, _, result, _)
   api.nvim_buf_add_highlight(headbuf, -1, 'Keyword', 0, -1)
 end
 
+M['textDocument/documentHighlight'] = function(_, _, result, _)
+  if not result then return end
+  local bufnr = api.nvim_get_current_buf()
+  for _, reference in ipairs(result) do
+    local start_pos = {reference["range"]["start"]["line"], reference["range"]["start"]["character"]}
+    local end_pos = {reference["range"]["end"]["line"], reference["range"]["end"]["character"]}
+    if reference["kind"] == 1 then
```

Comment:
`vim.lsp.protocol.DocumentHighlightKind` can be used instead of numbers + comments.

Reply:
fixed, thanks